lib/spack/spack/directives.py

```diff
@@ -758,6 +761,17 @@ def _execute_resource(pkg):
     return _execute_resource
 
 
+def build_system(*values, **kwargs):
```

Comment:
Needs a docstring. I would strongly recommend avoiding the use of `**kwargs`. It means that bugs like:
```python
build_system("cmake", "autotools", defalt="cmake")
```
go unnoticed. The only time `**kwargs` should ever be used in software is in wrapper functions that pass the entire dict to another function.

Reply:
>  I would strongly recommend avoiding the use of **kwargs

This cannot be done while we support Python 2.7 (we can't use keyword only arguments)